File: features/lender/src/main/java/com/nivasafinance/features/lender/lenderoffice/service/LenderOfficeWriteService.java
package com.nivasafinance.features.lender.lenderoffice.service;

import com.nivasafinance.features.lender.lenderoffice.dto.LenderOfficeReponseData;
import com.nivasafinance.features.lender.lenderoffice.dto.LenderOfficeRequestData;
import com.nivasafinance.features.lender.lenderoffice.dto.UpdateLenderOfficeRequest;

import java.util.UUID;

public interface LenderOfficeWriteService {
    LenderOfficeReponseData create(String lenderKey, LenderOfficeRequestData lenderOfficeData);
    LenderOfficeReponseData update(UUID id, LenderOfficeRequestData lenderOfficeData);
    LenderOfficeReponseData updateOffice(UUID id, UpdateLenderOfficeRequest request);
    void delete(UUID id);
    LenderOfficeReponseData activateDeactivateLenderOffice(UUID officeId);
}

File: features/lender/src/main/java/com/nivasafinance/features/lender/lenderoffice/service/LenderOfficeWriteService.kt
package com.nivasafinance.features.lender.lenderoffice.service

import com.nivasafinance.features.lender.lenderoffice.dto.LenderOfficeReponseData
import com.nivasafinance.features.lender.lenderoffice.dto.LenderOfficeRequestData
import java.util.UUID

interface LenderOfficeWriteService {
    fun create(lenderOfficeData: LenderOfficeRequestData): LenderOfficeReponseData
    fun update(id: UUID, lenderOfficeData: LenderOfficeRequestData): LenderOfficeReponseData
    fun delete(id: UUID)
}
